src/Tasks.UnitTests/WriteCodeFragment_Tests.cs

```diff
@@ -90,6 +90,33 @@ public void CombineFileDirectory()
             string file = Path.Combine(Path.GetTempPath(), "CombineFileDirectory.tmp");
             Assert.Equal(file, task.OutputFile.ItemSpec);
             Assert.True(File.Exists(file));
+
+            File.Delete(task.OutputFile.ItemSpec);
+        }
+
+        /// <summary>
+        /// Combine file and directory where the directory does not already exist
+        /// </summary>
+        [Fact]
+        public void CombineFileDirectoryAndDirectoryDoesNotExist()
```

Comment:
Using our `TestEnvironment` class will help streamline this even without going to a full modernization of the test logic. I'll push a small change.